From 4e495a37939a605577c72ed43e1f5a3ab3780611 Mon Sep 17 00:00:00 2001
From: babak sarashki <babak.sarashki@windriver.com>
Date: Sun, 3 Nov 2019 15:16:35 -0800
Subject: [PATCH 17/19] openldap bdb_idl_fetch_key correct key pointer

From Stx 1901: openldap-bdb_idl_fetch_key-correct-key-pointer.patch

Correct key pointer usage in bdb_idl_fetch_key.

Author: Shogo Matsumoto
Resolves: #1356165
---
 servers/slapd/back-bdb/idl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servers/slapd/back-bdb/idl.c b/servers/slapd/back-bdb/idl.c
index 3e0d7a6..7dae9ff 100644
--- a/servers/slapd/back-bdb/idl.c
+++ b/servers/slapd/back-bdb/idl.c
@@ -626,7 +626,7 @@ bdb_idl_fetch_key(
 					BDB_DISK2ID( j, i );
 				}
 			}
-			rc = cursor->c_get( cursor, key, &data, flags | DB_NEXT_DUP );
+			rc = cursor->c_get( cursor, kptr, &data, flags | DB_NEXT_DUP );
 		}
 		if ( rc == DB_NOTFOUND ) rc = 0;
 		ids[0] = i - ids;
-- 
2.17.1

